### PR TITLE
Kernel, part I

### PR DESCRIPTION
**Pull request by bennn**
The essentials from the `kernel/` folder of scmutils.
- [X] `interval.scm`
- [x] ~~enough support for symbolics to start working through~~ [the textbook](http://mitpress.mit.edu/sites/default/files/titles/content/sicm/book-Z-H-11.html#%_sec_1.4)
- [x] ~~basic struct definitions (up/down tuples)~~
- [x] move constants from `numeric.scm` to `main.rkt`

### CODE REVIEW

File: private/kernel/interval.rkt [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+    the-empty-interval
+    (interval lo hi lo-closed? hi-closed?)))
+
+;; We could possibly hack the #lang to distinguish square braces from parens.
+;; (#lang mechanics)
+;; TODO make sure unicode display correctly
+;; TODO use unicode << and [[ instead (double-angle and double-brace)
+(define-syntax (parse-interval stx)
+  ;; Do NOT check `lo:number`, let `make-interval/check` test & raise the error
+  (syntax-parse stx #:datum-literals (â¨ â© âª â«)
```
Comment (bennn) [inline private/kernel/interval.rkt]: lol, this is what I get for working with vim

File: private/kernel/interval.rkt [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+    [make-interval/check make-interval])
+  ;; Accessors
+  interval-lo
+  interval-hi
+  interval-lo-closed?
+  interval-hi-closed?
+  interval?
+  ;; Utilities
+  interval-empty?
+;  #:contract (-> interval? boolean?)
```
Comment (bennn) [inline private/kernel/interval.rkt]: Need to debug this, not sure what the syntax error with `provide/api` is.

**Comment by bkc39** [comment]
—

**Comment by bennn** [comment]
This is it for this pull request.
- The struct definitions aren't clear to me yet
- The printers are in the `display/` folder, and probably not important
- The literal functions (`litfun.scm`) don't seem like a good idea; I think switching to redex would be better

Next up: `numerics/quadrature/` is a pre-req for `mechanics/action.scm`.

**Comment by bennn** [comment]
Nvm, added `vector.rkt`. May add `matrix.rkt` also.

**Comment by bennn** [comment]
Not adding matrix.

**Pull request merged**